metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesFinder.scala

```diff
@@ -0,0 +1,109 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.meta.internal.implementation.ImplementationProvider
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.DefinitionProvider
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.debug.BuildTargetClasses
+
+import ch.epfl.scala.bsp4j.BuildTarget
+
+trait TestSuitesFinder {
```

Comment:
More of just a curiosity, there is only one implementation of this. Are there plans for another? Or was was the reason to provide this interface over just a single concrete class?

Reply:
There aren't plans for others implementations. 
Some time ago I played with this approach (trait + implementation) and I liked it. I find it very easy to reason about what code should do, because there is just a simple trait with all public API gathered -- without implementation overhead -- where you can quickly search & find methods. Sometimes, when there is so much code, quick look over API can tell more about class responsibilities than code. 

I know, it's a little bit strange to abstract single class, but I think it's worth to give this a try.

Reply:
👍🏼 cool,  thanks for explaining.

Reply:
Though, the drawback of this approach is that every time you use go to definition, you will later need to find implementations.

I am also not sure if it's less clear if a user looks at the implementation. It's exactly same line.

There is no benefit to seeing:
```scala
def findTestSuites(): Seq[TestSuiteDiscoveryResult]
```

separate from
```scala
override def findTestSuites(): Seq[TestSuiteDiscoveryResult] = {
    buildTargets.allBuildTargetIds.toList
    ....
```

I agree if there were to be a lot more methods, but with just one let's just have a class.

Reply:
> I am also not sure if it's less clear if a user looks at the implementation. It's exactly same line.

There are also constructor arguments so there is still some cognitive overhead, but I agree that in this case it's negligible and placing `override def`s at the beginning of the class is good enough.

Though I already have a feature branch with an additional method and I guess there's potential for two more :P

Reply:
Honestly, you can also just use `outline` in VS Code which allows a better view of what's going on in the class.

Reply:
Nah, it's something else. My problem was about `what is this class doing/what is public API` vs `how this class is doing it` and not about listing methods :/